File: 2032-largest-odd-number-in-string/largest-odd-number-in-string.cpp
class Solution {
public:
    string largestOddNumber(string num) {
        int n = num.size()-1;
        string ans = "";
        for(int i=n; i>-1; i--){
            if((num[i]-'48')%2!=0){
                ans = num.substr(0,i+1);
                break;
            }
        }
        return ans;
    }
};

File: 2032-largest-odd-number-in-string/largest-odd-number-in-string.java
class Solution {
    public String largestOddNumber(String num) {
        int i=0;
        for(i=num.length()-1; i>-1; i--){
            if((num.charAt(i)-'0')%2!=0){
                break;
            }
        }

        String ans = "";
        for(int j=0; j<=i; j++){
            ans = ans+num.charAt(j);
        }
        return ans;
    }
}